packages/react-accordion/src/components/AccordionHeader/useAccordionHeader.ts

```diff
@@ -53,9 +57,10 @@ export const useAccordionHeader = (
   const size = useContextSelector(AccordionContext, ctx => ctx.size);
   const id = useId('accordion-header-', props.id);
   const panel = useDescendants(accordionItemDescendantContext)[1] as AccordionItemDescendant | undefined;
+  const innerRef = React.useRef<HTMLElement>(null);
```

Comment:
can you remove the inner ref ? it's not being used for anything right ?

Reply:
oh wait, I see now, why was it hoisted out ?

Reply:
I've opted for hoisting it due to opting for `React.Ref` instead of `React.RefObject` from inner ref of state. I'm pretty much just following pattern I've found on `react-button` component